packages/material-ui-styled-engine-sc/package.json

```diff
@@ -0,0 +1,52 @@
+{
+  "name": "@material-ui/styled-engine-sc",
+  "version": "5.0.0-alpha.8",
+  "private": false,
+  "author": "Material-UI Team",
+  "description": "Material-UI Styled - Wrapper package for styled utility.",
+  "main": "./src/index.js",
+  "keywords": [
+    "react",
+    "react-component",
+    "material design",
+    "material-ui",
+    "styled"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/mui-org/material-ui.git",
+    "directory": "packages/material-ui-styled"
+  },
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/mui-org/material-ui/issues"
+  },
+  "homepage": "https://material-ui.com/styled/basics/",
+  "funding": {
+    "type": "opencollective",
+    "url": "https://opencollective.com/material-ui"
+  },
+  "scripts": {
+    "build": "yarn build:cjs && yarn build:esm && yarn build:es && yarn build:copy-files",
+    "build:cjs": "node ../../scripts/build cjs",
+    "build:esm": "node ../../scripts/build esm",
+    "build:es": "node ../../scripts/build es",
+    "build:copy-files": "node ../../scripts/copy-files.js",
+    "prebuild": "rimraf build",
+    "release": "yarn build && npm publish build --tag next",
+    "test": "cd ../../ && cross-env NODE_ENV=test mocha 'packages/material-ui-styles/**/*.test.{js,ts,tsx}' --exclude '**/node_modules/**'",
+    "typescript": "tslint -p tsconfig.json \"{src,test}/**/*.{spec,d}.{ts,tsx}\" && tsc -p tsconfig.json"
+  },
+  "dependencies": {},
+  "peerDependencies": {
+    "@types/styled-components": "5.1.3",
```

Comment:
Should be marked as an optional peer dependency via `peerDependenciesMeta`. Some of our packages already use `peerDependenciesMeta`

Reply:
So should both `@material-ui/styled-engine` and  `@material-ui/styled-engine-sc` add the emotions/styled-components dependencies as optional peer dependencies, and then we can have  `@material-ui/styled-engine` as a dependency in the core. Then if people want to use styled-components, they can use alias to point  `@material-ui/styled-engine` to  `@material-ui/styled-engine-sc`

Does this makes sense?

Reply:
Rationale for why some `@types/*` deps should be optional peers: https://github.com/mui-org/material-ui/pull/17211